### PR TITLE
keepalive: godoc improvement and behavior explanation

### PR DESCRIPTION
**Pull request by menghanl**

### CODE REVIEW

File: keepalive/keepalive.go [hunk 1]
```diff
@@ -16,50 +16,67 @@
...
+	// pings the server to see if the transport is still alive.
 	Time time.Duration // The current default value is infinity.
-	// After having pinged for keepalive check, the client waits for a duration of Timeout and if no activity is seen even after that
-	// the connection is closed.
+	// After having pinged for keepalive check, the client waits for a duration
+	// of Timeout and if no activity is seen even after that the connection is
+	// closed.
 	Timeout time.Duration // The current default value is 20 seconds.
-	// If true, client runs keepalive checks even with no active RPCs.
+	// If set to false, client doesn't run keepalive checks when there's no
```
Comment (dfawley) [inline keepalive/keepalive.go]: "If set to false" sounds to me like the default might be "true" (even though this is Go and we know that's not possible).  What was wrong with the old wording?

File: keepalive/keepalive.go [hunk 2]
```diff
@@ -16,50 +16,67 @@
...
+// EnforcementPolicy is used to set keepalive enforcement policy on the
+// server-side. Server will close connection with a client that violates this
+// policy.
 type EnforcementPolicy struct {
-	// MinTime is the minimum amount of time a client should wait before sending a keepalive ping.
+	// MinTime is the minimum amount of time a client should wait before sending
+	// a keepalive ping.
 	MinTime time.Duration // The current default value is 5 minutes.
-	// If true, server expects keepalive pings even when there are no active streams(RPCs).
+	// If true, server expects keepalive pings even when there are no active
```
Comment (dfawley) [inline keepalive/keepalive.go]: expects -> allows

File: keepalive/keepalive.go [hunk 3]
```diff
@@ -16,50 +16,67 @@
...
+	// streams(RPCs). If set to false, and client sends ping when there's no
```
Comment (dfawley) [inline keepalive/keepalive.go]: Nit: there's -> there are ("no active streams" is plural).

(There's another instance of this above in the client-side PermitWithoutStream.)

File: keepalive/keepalive.go [hunk 4]
```diff
@@ -16,50 +16,68 @@
...
+	// pings the server to see if the transport is still alive.
 	Time time.Duration // The current default value is infinity.
-	// After having pinged for keepalive check, the client waits for a duration of Timeout and if no activity is seen even after that
-	// the connection is closed.
+	// After having pinged for keepalive check, the client waits for a duration
+	// of Timeout and if no activity is seen even after that the connection is
+	// closed.
 	Timeout time.Duration // The current default value is 20 seconds.
-	// If true, client runs keepalive checks even with no active RPCs.
+	// If true, client runs keepalive checks even with no active RPCs. If false,
```
Comment (dfawley) [inline keepalive/keepalive.go]: "runs keepalive checks" -> "sends keepalive pings"?

Comment (menghanl) [reply keepalive/keepalive.go]: Done

**Comment by spzala** [comment]
@menghanl very helpful doc updates. Thanks!!

**Pull request merged**